File: CPP Language/CPP Language 2.0/ErrorHandeling.cpp
// cpp - error handeling
#include<iostream>
#include<exception>
using namespace std;
float Inverse(int x){
    if(x==0){
        // throws exception if 0
        throw("Undefined Value");
    }
    else return 1/x;
}
int divide(int x,int y){
    return x/y;
}
int main(){
    int num = 0;
    // try this code block
    try{
        cout<<Inverse(num);
    }
    // catch any exception if throwed
    catch(const char*e){
        cout<<"Error Occured ! first try";
    }

    try{
        cout<<divide(1,0) <<endl;
    }
    catch(const char*e){
        cout<<"Error Occured !";
    }
}

File: JavaScript Language/JavaScript Language/ErrorHandeling.js
    // javascript - Error Handeling
    function println(params){
        console.log(params);
    }
    println('Hello World');
    function div(a,b){
        return a/b;
    }
    try{
        // undeclared varibles
        a = b;
        println(a/b);
    }
    catch(error){
        println('Errore Occured !');
    }

    // example 2
    try{
        println('Try block which generate error ');
        println(div(a))
    }catch(error){
        println('This is catch block !\nexecutes when error occures')
    }finally{
        println('This is finally block !\nExecutes at any case')
    }
    
    function factorial(num){
        if(num<0) throw('Number less than 0 !');
        else{
            var result = 0
            while(num>0){
                result *= num;
                num--;
            }
            return result;
        }
    }

    // example 3
    try{
        var n1 = factorial(-1);
    }catch(error){
        println('error occured !'+error);
    }